### ISSUE TITLE
Solved: UnicodeDecodeError: 'ascii' codec can't decode byte error with latest build

### ISSUE THREAD

**Issue by jamesfrances** [kiwix/kiwix-build#313]
I had this python build error. It took some hunting around but I found if you did this first, it would build nicely.
      export LC_ALL="en_US.UTF-8"
      export LC_CTYPE="en_US.UTF-8"

**Comment by kelson42** [comment]
@mgautierfr a bug by us?

**Comment by kelson42** [comment]
@jamesfrances The full log would be helpful

**Comment by mgautierfr** [comment]
On what system are you running the build ?
What was the default values of LC_ALL and LC_CTYPE ?
Yes, the log (at least the error message and the trace) would be useful.

**Comment by jamesfrances** [comment]
it's being build on a 64bit raspberrypi clone. fresh install with minimal configuration

LC_ALL = C, LC_CTYPE is blank/null

# kiwix-build
[INSTALL PACKAGES]
 - automake : SKIP
 - libtool : SKIP
 - cmake : SKIP
 - git : SKIP
 - subversion : SKIP
 - ccache : SKIP
 - pkg-config : SKIP
 - gcc : SKIP
 - autopoint : SKIP
 - libbz2-dev : SKIP
 - libmagic-dev : SKIP
 - zlib1g-dev : SKIP
 - uuid-dev : SKIP
 - libmicrohttpd-dev : SKIP
SKIP, No package to install.
[SETUP PLATFORMS]
[PREPARE]
prepare sources pugixml :
  download pugixml : OK1.50%
  extract pugixml : OK
  patch pugixml : ERROR
Traceback (most recent call last):
  File "/usr/local/bin/kiwix-build", line 11, in <module>
    load_entry_point('kiwix-build==0.1', 'console_scripts', 'kiwix-build')()
  File "/usr/local/lib/python3.6/dist-packages/kiwixbuild/__init__.py", line 96, in main
    builder.run()
  File "/usr/local/lib/python3.6/dist-packages/kiwixbuild/builder.py", line 197, in run
    self.prepare_sources()
  File "/usr/local/lib/python3.6/dist-packages/kiwixbuild/builder.py", line 107, in prepare_sources
    source.prepare()
  File "/usr/local/lib/python3.6/dist-packages/kiwixbuild/dependencies/base.py", line 123, in prepare
    self.command('patch', self._patch)
  File "/usr/local/lib/python3.6/dist-packages/kiwixbuild/dependencies/base.py", line 75, in command
    ret = function(*args, context=context)
  File "/usr/local/lib/python3.6/dist-packages/kiwixbuild/dependencies/base.py", line 68, in _patch
    run_command("patch -p1", self.source_path, context, input=patch_input.read())
  File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 10: ordinal not in range(128)

**Comment by kelson42** [comment]
@mgautierfr It seems at the core there is a pugixml patch error!

**Comment by michaelsilver** [comment]
I experienced the same `UnicodeDecodeError` error running `kiwix-build kiwix-lib` or `kiwix-build zim-tools` on Ubuntu 16.04.6 LTS. I found the issue was caused by my machine not having its locale  properly set (see this [StackOverflow post](https://stackoverflow.com/questions/51026315/how-to-solve-unicodedecodeerror-in-python-3-6/51027262#51027262)).

To check if your machine also doesn't have its locale properly set, run `locale` in the terminal. The **correct** output looks like this (for United States, English):
```
$ locale
LANG=en_US.utf8
LANGUAGE=
LC_CTYPE="en_US.utf8"
LC_NUMERIC="en_US.utf8"
LC_TIME="en_US.utf8"
LC_COLLATE="en_US.utf8"
LC_MONETARY="en_US.utf8"
LC_MESSAGES="en_US.utf8"
LC_PAPER="en_US.utf8"
LC_NAME="en_US.utf8"
LC_ADDRESS="en_US.utf8"
LC_TELEPHONE="en_US.utf8"
LC_MEASUREMENT="en_US.utf8"
LC_IDENTIFICATION="en_US.utf8"
LC_ALL=
```

If your variables aren't set to `en_US.utf8`, check out [these instructions](https://chat.stackoverflow.com/rooms/173761/discussion-between-alastair-mccormack-and-umair) (linked from the above StackOverflow answer) to fix your machine's locale settings.
